pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+package sso
+
+import (
+	"context"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services"
+	"github.com/golang/glog"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+	"time"
+)
+
+type redhatssoService struct {
+	client redhatsso.SSOClient
+}
+
+func (r *redhatssoService) RegisterKafkaClientInSSO(accessToken string, kafkaNamespace string, orgId string) (string, *errors.ServiceError) {
+	// TODO: implement
+	return "", nil
+}
+
+func (r *redhatssoService) RegisterOSDClusterClientInSSO(accessToken string, clusterId string, clusterOathCallbackURI string) (string, *errors.ServiceError) {
+	// TODO: implement
+	return "", nil
+}
+
+func (r *redhatssoService) DeRegisterClientInSSO(accessToken string, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil {
+		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, "failed to delete the sso client")
+	}
+	glog.V(5).Infof("Kafka Client %s deleted successfully", clientId)
+	return nil
+}
```

Comment:
@akoserwal old api was deleting the client by using an 'internalid'. Is it ok to simply delete by `clientId` with the new API?

Reply:
How this access token will be passed?

yes, you can use `clientId`